### ISSUE TITLE
linter.verify does not validate rule configs

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#8325]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 7.7.4
* **npm Version:** 4.1.2

**What did you do? Please include the actual source code causing the issue.**

```js
require("eslint").linter.verify("foo", {
    rules: {
        "no-restricted-properties": ["error", "foo"]
    }
});
```

**What did you expect to happen?**

I expected core to throw a config validation error. Note that `["foo"]` is not a valid config for the `no-restricted-properties` rule, which accepts a list of objects.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint returns an empty array (i.e. reports no problems). The `no-restricted-properties` rule gets run with a string option, violating its schema invariant. (`no-restricted-properties` happens to handle this gracefully and report no errors, but some other rules crash when this happens.)

I'm pretty sure this is a bug -- although `linter` doesn't get configs from the filesystem, it seems like it should still preserve invariants for rules by throwing an error rather than having undefined behavior when a config is invalid.

**Comment by ilyavolodin** [comment]
I'm not sure it would be a good idea to validate config in `linter`. Doing that would validate config for every single file that you lint, vs. validating actual config. I think this could affect performance in a significant way.

**Comment by platinumazure** [comment]
In addition to @ilyavolodin's comments, I also feel that config validation would violate single responsibility principle for the linter object and so it wouldn't make sense architecturally.

**Comment by not-an-aardvark** [comment]
It seems like this is working as intended, so I'll close it. I had opened it because I thought it was unintended behavior (seems like we shouldn't expose an API that allows rules to be passed an invalid schema).
